bundles/org.corpus_tools.hexatomic.graph/src/main/java/org/corpus_tools/hexatomic/graph/internal/AnnotationFilterWidget.java

```diff
@@ -0,0 +1,195 @@
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package org.corpus_tools.hexatomic.graph.internal;
+
+import static org.eclipse.jface.widgets.WidgetFactory.text;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import org.corpus_tools.hexatomic.styles.ColorPalette;
+import org.corpus_tools.salt.common.SDocumentGraph;
+import org.eclipse.draw2d.ColorConstants;
+import org.eclipse.e4.core.services.events.IEventBroker;
+import org.eclipse.jface.fieldassist.ContentProposalAdapter;
+import org.eclipse.jface.fieldassist.IContentProposal;
+import org.eclipse.jface.fieldassist.IContentProposalListener;
+import org.eclipse.jface.fieldassist.TextContentAdapter;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.jface.layout.RowDataFactory;
+import org.eclipse.jface.layout.RowLayoutFactory;
+import org.eclipse.nebula.widgets.chips.Chips;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.ScrolledComposite;
+import org.eclipse.swt.events.ControlAdapter;
+import org.eclipse.swt.events.ControlEvent;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Text;
+
+/**
+ * A composite that allows to define filter critera.
+ * 
+ * @author Thomas Krause {@literal thomas.krause@hu-berlin.de}
+ */
+public class AnnotationFilterWidget extends Composite
+    implements IContentProposalListener, AnnotationFilter {
+
+  public static final String ANNO_FILTER_CHANGED_TOPIC = "GRAPH_EDITOR/ANNOTATION_FILTER/CHANGED";
+
+  private final Text txtAddAnnotatioName;
+
+  private final AnnotationNameProposalProvider proposalProvider;
+
+  private final List<Chips> activeChips = new LinkedList<>();
+  private final Set<String> filteredNames = new TreeSet<>();
+
+  private final Composite facetFilterComposite;
+
+
+  private final IEventBroker eventBroker;
+
+  private ScrolledComposite scroll;
+
+  /**
+   * Create a new filter widget.
+   * 
+   * @param parent The SWT parent composite
+   * @param saltGraph The document graph that will be filtered. Needed to extract the annotation
+   *        names.
+   * @param eventBroker Is used to post events whenever there is an update on the filter model.
+   */
+  public AnnotationFilterWidget(Composite parent, SDocumentGraph saltGraph,
+      IEventBroker eventBroker) {
+    super(parent, SWT.BORDER);
+    this.eventBroker = eventBroker;
+    this.setLayout(GridLayoutFactory.swtDefaults().create());
+
+    txtAddAnnotatioName = text(SWT.BORDER)
```

Comment:
```suggestion
    txtAddAnnotationName = text(SWT.BORDER)
```

Fix typo in this variable name (in all instances of course ;)).

Reply:
Changed in 2151cbac